jenkins/end_to_end_tests.sh

```diff
@@ -17,7 +17,14 @@ virtualenv venv
 
 pip install -r requirements/base.txt
 
-# Install pages.
+# Install the page objects from the edx-platform repo.
+# Before doing so, we don't need optimizations for lxml,
+# so install it this way which doesn't bother compiling them.
+STATIC_DEPS=true CFLAGS="-O0"  pip install "lxml==3.4.4"
 paver install_pages
+
+# Set the display to the virtual frame buffer (Xvfb)
+export DISPLAY=:1
```

Comment:
Reviewer note: This was why firefox was erroring instead of starting, it had nowhere to draw the display.
In the edx-platform repo jobs, this is handled by the [jenkins-common script](https://github.com/edx/edx-platform/blob/master/scripts/jenkins-common.sh#L3), which sources [a file that was put in the jenkins user's home directory](https://github.com/edx/configuration/blob/master/playbooks/roles/jenkins_worker/templates/jenkins_env.j2#L5) on the jenkins-worker during its initial configuration.